svcdef/svcdef.go

```diff
@@ -415,6 +417,14 @@ func NewField(f *ast.Field) (*Field, error) {
 	// both, and modifying the return value accordingly.
 	var typeFollower func(ast.Expr) error
 	typeFollower = func(e ast.Expr) error {
+		if f.Tag != nil {
+			r := regexp.MustCompile(`json:"(.+),`)
+			m := r.FindStringSubmatch(f.Tag.Value)
+			if len(m) > 1 {
```

Comment:
If `protoc-gen-go` changes what order it writes the struct tags, or if json is used again in the struct tag, this could be brittle using `regexp`.

For example in an old version of `protoc-gen-go` the struct tags looks like this:
```
protobuf:"bytes,32,rep,name=BBB" json:"BBB,omitempty" protobuf_key:"varint,1,opt,name=key" protobuf_val:"fixed32,2,opt,name=value
```
Which is in some of the `svcdef` tests.

Your code parses the `PBFieldName` to be the following:
```
BBB,omitempty" protobuf_key:"varint,1,opt,name=key" protobuf_val:"fixed32,2,opt
```

Which is not desired.

We could use `reflect` to parse the `StructTags`. I looked at `encoding/json` and put this little example together that seems to work.

```
if f.Tag != nil {
	tag := reflect.StructTag(f.Tag.Value).Get("json")
	if idx := strings.Index(tag, ","); idx != -1 {
		rv.PBFieldName = tag[:idx]
	}
}
```

This should avoid the issues of using regex to parse these.